### PR TITLE
[KYUUBI #4144][FOLLOWUP] Do not cleanup upload dir because of batch recovery and fix temp file leak

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Address comments: https://github.com/apache/kyuubi/pull/4144#issuecomment-1412078077
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
Seems need to cleanup when closing batch op to prevent temp file leak. Will address it later.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala [hunk 1]
```diff
@@ -363,6 +363,12 @@ class BatchJobSubmission(
   override def isTimedOut: Boolean = false
 
   override protected def eventEnabled: Boolean = true
+
+  private def cleanupUploadedResourceIfNeeded(): Unit = {
+    if (session.isResourceUploaded && new File(resource).exists()) {
```
Comment (bowenliang123) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala:368]: How about skip check the existence of the file, and then make `Utils.deleteFile` use `Files.deleteIfExists` instead?
```suggestion
    if (session.isResourceUploaded) {
```

**Review by bowenliang123 — APPROVED** [review]
LGTM. Minor suggestion only.

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**